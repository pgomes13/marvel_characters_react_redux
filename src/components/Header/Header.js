import React from 'react';
import PropTypes from 'prop-types';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from "@material-ui/core/Toolbar/Toolbar";
import CameraIcon from '@material-ui/icons/PhotoCamera';
import Typography from "@material-ui/core/Typography/Typography";
import { withStyles } from '@material-ui/core/styles';

import { HeaderStyles } from './Header.styles';

/**
 * The Header component consisting of the header info
 * @param {Object} classes - the material-ui classes prop
 * @param {String} title - the title prop
 * @returns {Node} - the Header component
 */
const _Header = ({ classes, title }) => (
    <AppBar position="static" className={classes.appBar}>
        <Toolbar>
            <CameraIcon className={classes.icon} />
            <Typography variant="h6" color="inherit" noWrap>
                {title}
            </Typography>
        </Toolbar>
    </AppBar>
);

_Header.propTypes = {
    classes: PropTypes.object.isRequired,
    title: PropTypes.string
};

const Header = withStyles(HeaderStyles)(_Header);

export { Header };
