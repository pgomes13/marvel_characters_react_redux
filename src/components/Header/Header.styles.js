const HeaderStyles = theme => ({
    appBar: {
        position: 'relative',
    },
    icon: {
        marginRight: theme.spacing.unit * 2,
    }
});

export { HeaderStyles };