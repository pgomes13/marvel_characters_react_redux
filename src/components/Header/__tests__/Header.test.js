import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { _Header } from '../Header';

describe('Header', () => {
    const props = {
        classes: {},
        title: 'test',
    };

    it('component renders correctly with props', () => {
        const wrapper = shallow(<_Header {...props} />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});