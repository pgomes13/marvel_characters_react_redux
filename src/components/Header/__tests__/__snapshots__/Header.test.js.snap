// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly with props 1`] = `
<WithStyles(AppBar)
  position="static"
>
  <WithStyles(Toolbar)>
    <pure(PhotoCameraIcon) />
    <WithStyles(Typography)
      color="inherit"
      noWrap={true}
      variant="h6"
    >
      test
    </WithStyles(Typography)>
  </WithStyles(Toolbar)>
</WithStyles(AppBar)>
`;
